model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/group/entity/JpaGroupAttributeEntity.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.models.map.storage.jpa.group.entity;
+
+import java.util.Objects;
+import java.util.UUID;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import org.hibernate.annotations.Nationalized;
+import org.keycloak.models.map.storage.jpa.JpaChildEntity;
+
+@Entity
+@Table(name = "group_attribute")
+public class JpaGroupAttributeEntity implements JpaChildEntity<JpaGroupEntity> {
```

Comment:
Wondering if the attribute entity would benefit from a common parent.

Reply:
I was thinking about it and didn't come up with any benefits so far. Do you have some specific ones in mind?

Reply:
The structure seems always the same. If we rather than `fk_group` used some neutral name like `fk_area` (similarly for setter / getter), one could introduce a parent `JpaAttributeEntity<E> extends JpaChildEntity<E>` that would contain all the logic and the descendant would be like this

```java
@Entity
@Table(name = "group_attribute")
public class JpaGroupAttributeEntity extends JpaAttributeEntity<JpaGroupAttributeEntity> {

}
```
Note the empty class body

Reply:
Thank you. I've created an issue for other areas: https://github.com/keycloak/keycloak/issues/10071